File: Data Structures & Algorithms/buy-and-sell-crypto/submission-3.js
class Solution {
    /**
     * @param {number[]} prices
     * @return {number}
     */
    maxProfit(prices) {
        var maxProfit = 0;
        var minBuy = prices[0];
        for (var i = 0; i < prices.length; i++) {
            maxProfit = Math.max(maxProfit, prices[i] - minBuy);
            minBuy = Math.min(minBuy, prices[i]);
        }
        return maxProfit;
    }
}

File: Data Structures & Algorithms/max-water-container/submission-3.java
class Solution {
    public int maxArea(int[] heights) {
        int left = 0;
        int right = heights.length -1;
        int water = 0;
        while (left < right) {
            int minH = Math.min(heights[left], heights[right]);
            int distance = right - left;
            int currWater = minH * distance;
            water = Math.max(currWater, water);
            if (heights[left] < heights[right]) left++;
            else right--;
        }
        return water;
    }
}
